lib/util/Components.js

```diff
@@ -402,7 +402,7 @@ function componentRule(rule, context) {
       const propertyNames = ['forwardRef', 'memo'];
       const calleeObject = node.callee.object;
       if (calleeObject) {
-        return arrayIncludes(propertyNames, node.callee.property.name) && node.callee.object.name === pragma;
```

Comment:
This isn’t equivalent, because the first one has .property and not .object.

Reply:
Maybe I should add check for both node.callee.property and node.callee.object?

Reply:
I’m not really sure - a regression test would help make that clear.

Reply:
Added a regression test